src/brevitas/ptq_algorithms/channel_splitting.py

```diff
@@ -0,0 +1,255 @@
+import math
+from typing import Dict, List, Set, Tuple, Union
+import warnings
+
+import torch
+import torch.nn as nn
+
+from brevitas.fx import GraphModule
+from brevitas.graph.base import GraphTransform
+from brevitas.graph.equalize import _batch_norm
+from brevitas.graph.equalize import _channel_maxabs
+from brevitas.graph.equalize import _extract_regions
+from brevitas.graph.equalize import _get_input_axis
+from brevitas.graph.equalize import _get_output_axis
+from brevitas.graph.equalize import Region
+from brevitas.graph.equalize import transpose
+
+_conv = (
+    nn.Conv1d, nn.Conv2d, nn.Conv3d, nn.ConvTranspose1d, nn.ConvTranspose2d, nn.ConvTranspose3d)
+
+
+def _channels_to_split(
+        sources: Dict[str, nn.Module],
+        sinks: Dict[str, nn.Module],
+        split_criterion: str,
+        split_ratio: float,
+        split_input: bool) -> Dict[nn.Module, List[torch.Tensor]]:
+    """
+    This method computes the channels that will be split based on `split_criterion`.
+    """
+    modules = sinks if split_input else sources
+    _get_axis = _get_input_axis if split_input else _get_output_axis
+    # the modules are all of the same shape so we can just take the first one
+    single_module = next(iter(modules.values()))
+    num_channels = single_module.weight.shape[_get_axis(single_module)]
+    splits_per_layer = int(math.ceil(split_ratio * num_channels))
+
+    module_to_channels = {}
+    if split_criterion == 'maxabs':
+        for name, module in modules.items():
+            # get input/output axis of module
+            axis = _get_axis(module)
+            # transpose to have axis as first dimension
+            weight_t = transpose(module, axis)
+            # flatten all but first dimension and get max per channel
+            max_per_channel = _channel_maxabs(weight_t.reshape(weight_t.size(0), -1))
+            channels_sorted = torch.argsort(max_per_channel, descending=True)
+            module_to_channels[name] = channels_sorted[:splits_per_layer]
+
+    # return tensor with the indices to split
+    channels_to_split = torch.cat(list(module_to_channels.values()))
+    return torch.unique(channels_to_split)
+
+
+def transpose_tensor(tensor: torch.Tensor, axis: int):
+    """
+    Given a tensor and an axis, this function re-arranges the tensor dimensions so that the axis and
+    the first dimension are swapped.
+    """
+    shape = list(range(tensor.ndim))
+    axis = shape[axis]
+    shape.insert(0, axis)
+    del shape[axis + 1]
+    return tensor.permute(shape)
+
+
+def _split_channels(module, channels_to_split, split_input=False, split_factor=0.5) -> None:
+    """
+    Splits the channels `channels_to_split` of the `weights`.
```

Comment:
The docstring is a bit weird. Maybe rewording.
Also if you add the Return keyword, then we might need to specify the entire function signature. Remove it for now